api/token.go

```diff
@@ -629,11 +654,10 @@ func (a *API) issueRefreshToken(ctx context.Context, conn *storage.Connection, u
 	if err != nil {
 		return nil, err
 	}
-
 	return &AccessTokenResponse{
 		Token:        tokenString,
 		TokenType:    "bearer",
-		ExpiresIn:    config.JWT.Exp,
+		ExpiresIn:    int(currentClaims.StandardClaims.ExpiresAt - time.Now().Unix()),
```

Comment:
why don't we use the JWT exp here? won't the expiresAt - timeNow == jwt expiry?

Reply:
oh in this scenario we are taking an existing JWT and adding the additional MFA Claims etc here. Since we are modifying an existing JWT and not re-issuing a JWT, time has lapsed since we issued the JWT and the duration till expiry will be different. 

Concretely,

1500 -> User signs in, Token issued
1505 -> User successfully verifies, claims are added till token but `ExpiresIn` should have been reduced by 5 mins

It's admittedly not clean and I'm open to other suggestions if anyone has any

Reply:
hmm i actually think it's ok for the expiry to be the jwt exp because the existing JWT is being upgraded to an AAL2 jwt right? so it's technically not the same jwt as before

cc @hf

Reply:
Am okay either way

Reply:
I get the logic, but let's just use Occam's razor and use the simpler of the two solutions: using the default JWT expiry time. There can be subtle errors with the calculation (like getting a 0 or negative number) that need to be handled elsewhere.

I don't see anything extra that can be gained by adding the calculation, though certainly is very logical.